### ISSUE TITLE
Accessibility

### ISSUE THREAD

**Issue by salomerohr** [CERNDocumentServer/cds-ils#863]
**Is your feature request related to a problem? Please describe.**

The Diversity & Inclusion Team made the recommendations to be compliant in terms of accessibility.

Here are 2 reports highlighting the current issues with the catalogue:

https://wave.webaim.org/report#/https://catalogue.library.cern/ 

[catalogue.library.cern-03.11.2024.pdf](https://github.com/CERNDocumentServer/cds-ils/files/14557818/catalogue.library.cern-03.11.2024.pdf)

**Comment by JakobMiesner** [comment]
problem: [Semantic-UI seems to not support "Keyboard accessibility and other WCAG requirements"](https://github.com/Semantic-Org/Semantic-UI-React/issues/3745)

**Comment by JakobMiesner** [comment]
The PRs above focus on making the ILS frontside more usable with a11y tools (e.g. screenreaders and keyboard). The [WAVE tool](https://wave.webaim.org/) no longer shows errors on the pages mentioned in the [main PR](https://github.com/inveniosoftware/react-invenio-app-ils/pull/707) with the exception of:
1. contrast errors. 
This should be rather easy to fix but is a style decision. WAVE complains about multiple cases where we have low contrast (e.g. white text on orange background)

2. "Missing form label" errors in the search (see image). 
This is a more deeply rooted problem coming from react searchkit -> Semantic UI. 
Semantic UI seems to have pretty bad a11y support as mentioned [above](https://github.com/CERNDocumentServer/cds-ils/issues/863#issuecomment-3985218452).
The problem: the checkboxes in filters to the left in the search do not point to the label that labels them. Optimally (as far as I understand) they would use `aria-labeledby` and point at the label.
RDM seems to have solved this by just placing `aria-label` on the checkboxes themselves. This makes the search usable but still leaves the labels "orphaned".

A similar problem exists for Popups. The main PR "solves" this by adding an `aria-label` to the help icon. Using `aria-describedby` would be cleaner but I could not get it to work.

<img width="1342" height="1357" alt="Image" src="https://github.com/user-attachments/assets/5b2635b6-b61c-4bea-b507-edbd026a84ff" />
